### PR TITLE
feat: #148: Actor created message to federation

### PR DESCRIPTION
**Pull request by devanbenz**
- Implements Actor model and enums 
- Sends Actor message to consumer for federation depending on whether federation is enabled

### CODE REVIEW

File: docs/docker-compose.yml [hunk 1]
```diff
@@ -56,6 +56,7 @@ services:
       FEDERATION_QUEUE_USER: sublinks
       FEDERATION_QUEUE_PASS: sublinks
       FEDERATION_QUEUE_NAME: federation
+      FEDERATION_ROUTING_KEY: federation
```
Comment (lazyguru) [inline docs/docker-compose.yml:59]: Routing key needs to be hard-coded and will vary based on object sent. See https://github.com/sublinks/sublinks-federation/pull/43/files#diff-058eae10080651784c26b74b93f5037b2d0d58736861057b3eb9c0b217eca334R51

File: src/main/java/com/sublinks/sublinksapi/federation/enums/ActorType.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.sublinks.sublinksapi.federation.enums;
+
+public enum ActorType {
+  USER("user"),
+  COMMUNITY("community");
```
Comment (lazyguru) [inline src/main/java/com/sublinks/sublinksapi/federation/enums/ActorType.java:5]: I'm using Person and Group to match activity pub types (haven't pushed that code yet). However, I'm open to discussion on if we stick with user/community instead

**Review by lazyguru — CHANGES_REQUESTED** [review]

File: src/main/java/com/sublinks/sublinksapi/federation/enums/ActorType.java (hunk 2 above)
Comment (lazyguru) [reply src/main/java/com/sublinks/sublinksapi/federation/enums/ActorType.java:5]: I take it back, I did push some code on this. I started created JSON Schema files: https://github.com/sublinks/sublinks-federation/pull/43/files#diff-5d252b8ffdf59c8a7c138bfbb9b0d43075698195dc3cadba0c295641c387be2a

File: docs/docker-compose.yml (hunk 1 above)
Comment (devanbenz) [reply docs/docker-compose.yml:59]: Whoops! Let me rename this var it should be the exchange and not the routing key

**Review by lazyguru — APPROVED** [review]
lgtm

**Review by lazyguru — APPROVED** [review]

**Pull request merged**